### PR TITLE
Contact Struct

### PR DESCRIPTION
**Pull request by neekolas**
## Summary

I can see some conflicts brewing with @jazzz's recent changes. Wanted to break this piece out of my conversation initiation work before the conflicts get worse.

### CODE REVIEW

File: xmtp/src/client.rs [hunk 1]
```diff
@@ -20,6 +20,12 @@ pub enum Network {
 
 #[derive(Debug, Error)]
 pub enum ClientError {
+    #[error("contact error {0}")]
+    Contact(#[from] ContactError),
+    #[error("could not publish: {0}")]
+    PublishError(String),
+    #[error("Query failed: {0}")]
+    QueryError(String),
```
Comment (jazzz) [inline xmtp/src/client.rs:28]: 😎

File: xmtp/src/client.rs [hunk 2]
```diff
@@ -122,17 +127,19 @@ mod tests {
             .unwrap();
 
         assert_eq!(contacts.len(), 1);
-        assert!(contacts[0].prekey.is_some());
-        assert!(contacts[0].identity_key.is_some());
+        contacts[0].vmac_identity_key();
+        contacts[0].vmac_fallback_key();
```
Comment (jazzz) [inline xmtp/src/client.rs:131]: what are we hoping to check here?

File: xmtp/src/utils.rs [hunk 3]
```diff
@@ -21,3 +22,7 @@ pub fn build_envelope(content_topic: String, message: Vec<u8>) -> Envelope {
         timestamp_ns: get_current_time_ns(),
     }
 }
+
+pub fn base64_encode(bytes: &[u8]) -> String {
+    general_purpose::STANDARD_NO_PAD.encode(bytes)
```
Comment (jazzz) [inline xmtp/src/utils.rs:27]: Is there a reason to default to no padding?

**Review by jazzz — APPROVED** [review]
(👍, 🚢 )

File: xmtp/src/client.rs (hunk 2 above)
Comment (neekolas) [reply xmtp/src/client.rs:131]: That it doesn't panic. There are a bunch of layers of unwraps happening under the hood

File: xmtp/src/utils.rs (hunk 3 above)
Comment (neekolas) [reply xmtp/src/utils.rs:27]: Did not give it a ton of thought. I just prefer NO_PAD aesthetically

**Pull request merged**